src/Microsoft.Diagnostics.Monitoring.WebApi/DacLocator.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    internal static class DacLocator
+    {
+        private sealed class Runtime
+        {
+            public string Name { get; set; }
+
+            public Version Version { get; set; }
+
+            public string Path { get; set; }
+
+            public string GetRuntimeDirectory() => System.IO.Path.Combine(Path, Version.ToString());
+        }
+
+        public static async Task<string> LocateRuntime(CancellationToken token)
+        {
+            ProcessStartInfo processStartInfo = new ProcessStartInfo()
+            {
+                CreateNoWindow = true,
+                Arguments = "--list-runtimes",
+                UseShellExecute = false,
+                FileName = "dotnet",
+                RedirectStandardOutput = true,
+            };
+            //CONSIDER We could get the location of the runtime dll for a specific process by collecting rundown instead.
```

Comment:
Note this doesn't work across sidecars at all and will produce the wrong result. Consider guarding against that scenario.